geoh5py/objects/geo_image.py

```diff
@@ -380,12 +382,16 @@ def image(self, image: str | np.ndarray | BytesIO | Image.Image):
                 f"Get type {type(image)} instead."
             )
 
-        with TemporaryDirectory() as tempdir:
-            ext = getattr(image, "format")
-            temp_file = (
-                Path(tempdir) / f"image.{ext.lower() if ext is not None else 'tiff'}"
+        if image.mode not in ["RGBA", "RGB", "L"]:
```

Comment:
I think "CMYK" would be supported as well

Reply:
No it won't. 
PNG files does not support it. 

A workaround could be to manage the creation of a jpeg if mode = "CMYK".
However, it takes other compression options, and that's more code. 

Which option do we choose?

Reply:
Create a dictionary to manage compression options